### PR TITLE
Fix null being passed as OK button text in the lock dialog

### LINKED ISSUE #345: Silent error while executing Lock/unlock command
Right-click on any item in solution -> Select context menu command TFVC - Lock/unlock

Progress bar dialog shows up and then error occurs:

`java.lang.IllegalArgumentException: Argument for @NotNull parameter 'text' of com/intellij/openapi/ui/DialogWrapper.setOKButtonText must not be null
	at com.intellij.openapi.ui.DialogWrapper.$$$reportNull$$$0(DialogWrapper.java)
	at com.intellij.openapi.ui.DialogWrapper.setOKButtonText(DialogWrapper.java)
	at com.microsoft.alm.plugin.idea.common.ui.common.BaseDialogImpl.<init>(BaseDialogImpl.java:51)
	at com.microsoft.alm.plugin.idea.tfvc.ui.LockItemsDialog.<init>(LockItemsDialog.java:50)
	at com.microsoft.alm.plugin.idea.tfvc.actions.LockAction.execute(LockAction.java:46)
	at com.microsoft.alm.plugin.idea.tfvc.actions.MultipleItemAction.actionPerformed(MultipleItemAction.java:130)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:280)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:296)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:281)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:77)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:285)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:112)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:94)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:112)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:517)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:36)
	at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:539)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6651)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
	at java.desktop/java.awt.Component.processEvent(Component.java:6416)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5026)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:974)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:912)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:844)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:739)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:502)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)`

### PR DESCRIPTION
**Pull request by ForNeVeR**
This dialog has no OK button and tries to pass null as a corresponding text. The setter method shouldn't be called in this case.

Closes #345.

### CODE REVIEW

File: plugin/src/com/microsoft/alm/plugin/idea/common/ui/common/BaseDialogImpl.java [hunk 1]
```diff
@@ -40,15 +40,22 @@ public BaseDialogImpl(final Project project, final String title, final String ok
...
         this.showFeedback = showFeedback;
         this.feedbackContext = feedbackContext;
         this.project = project;
-        this.properties = properties != null ? new HashMap<String, Object>(properties) : Collections.<String, Object>emptyMap();
+        this.properties = properties != null ? new HashMap<>(properties) : Collections.emptyMap();
 
         super.setTitle(title);
-        super.setOKButtonText(okButtonText);
+        if (okButtonText != null)
+            super.setOKButtonText(okButtonText);
```
Comment (mfilippov) [inline plugin/src/com/microsoft/alm/plugin/idea/common/ui/common/BaseDialogImpl.java:58]: Maybe we should set the default value in this case? Or button already has default value?

Comment (ForNeVeR) [reply plugin/src/com/microsoft/alm/plugin/idea/common/ui/common/BaseDialogImpl.java:58]: It already has a sensible default (just "OK"), so it is totally okay for us to not call that method at all.

**Comment by mfilippov** [comment]
@ForNeVeR CI failed

**Comment by ForNeVeR** [comment]
It was a network failure while downloading the dependencies. I've restarted it and, it has been finished successful.

**Pull request merged**